doc/performance.md

```diff
@@ -1,39 +1,257 @@
 # Performance
 
-There is a Vagrantfile with performance inspection tools like SystemTap.
+- [**General guidelines**](#general-guidelines)
+- [**Default caching**](#default-caching)
+- [**Async reporting threads**](#async-reporting-threads)
+- [**Batching policy**](#batching-policy)
 
-```shell
-vagrant up
-vagrant ssh
-```
+This document gives some general guidelines to debug performance issues in
+APIcast. It also introduces the caching modes available and explains how they
+can help in increasing performance.
 
-APIcast is mounted into `/home/vagrant/app` so you can start it by:
+## General guidelines
 
-```shell
-cd app
-bin/apicast
-```
+When experiencing performance issues in APIcast, the first thing that we
+recommend to do is to try to determine the component that is responsible for
+those issues. In a typical APIcast deployment, there are three components to
+take into account:
 
-For profiling with stapxx it is recommended to start just one process and worker:
+- APIcast
+- The 3scale backend that authorizes requests and keeps track of the usage
+- The upstream API
 
-```shell
-bin/apicast -c examples/configuration/echo.json  > /dev/null
-```
+In order to measure the latency that APIcast plus the 3scale backend introduce,
+the first thing that we need to do is to measure the latency of the upstream
+API. There are many tools that can be used for that:
```

Comment:
In general, we avoid linking to third-party products in the main body of the docs. If mentioning these products is essential, you can create a separate section.

Reply:
I think it's fine. Those are open-source projects. In a product documentation I can see how this could become confusing regarding the level of support and other things, but in an upstream repo, I think it should be OK.

Reply:
It's a matter of maintainability. If we have an "additional resources" section, we can easily locate and update links and any other content related to third-party components.
On the other hand, given the current state of technology, open source projects can change their licensing features, requiring us to quickly make necessary updates to avoid legal issues.